OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -826,6 +826,34 @@ internal static void UpgradeWeaponRules(int engineVersion, ref List<MiniYamlNode
 					}
 				}
 
+				// Refactor Missile RangeLimit from ticks to WDist
+				if (engineVersion < 20160501)
+				{
+					var weapRange = node.Value.Nodes.FirstOrDefault(n => n.Key == "Range");
+					var projectile = node.Value.Nodes.FirstOrDefault(n => n.Key == "Projectile");
+
+					if (projectile != null && weapRange != null && projectile.Value.Value == "Missile")
+					{
+						var oldWDist = FieldLoader.GetValue<WDist>("Range", weapRange.Value.Value);
+						var rangeLimitNode = projectile.Value.Nodes.FirstOrDefault(x => x.Key == "RangeLimit");
+
+						// RangeLimit is now a WDist value, so for the conversion, we take weapon range and add 20% on top.
+						// Overly complicated calculations using Range, Speed and the old RangeLimit value would be rather pointless,
+						// because currently most mods have somewhat arbitrary, usually too high and in a few cases too low RangeLimits anyway.
+						var newValue = oldWDist.Length * 120 / 100;
+						var newCells = newValue / 1024;
+						var newCellPart = newValue - (newCells * 1024);
```

Comment:
Let's say `oldWDist` was `2c0 / 2048`.

`newValue` is the full new distance value, in our example `2457` (`2c409`).
`newCells` divides that by 1024, the result would be ~2.4 in fp, but is truncated to 2 for integer div, so this is the number of full cells (in other words, the number before the `c)`.
`newCellPart` then takes the original full WDist (2457), and substracts the number of full cells * 1024 (2 * 1024 = 2048 in this example), so the result here will be 2457 - 2048 = 409. So we have the WDist to put after the `c` as well now.